pynecone/var.py

```diff
@@ -773,72 +773,155 @@ def __init__(
+    def append(self, *args, **kwargs):
         """Append.
 
         Args:
             args: The args passed.
-            kargs: The kwargs passed.
+            kwargs: The kwargs passed.
         """
-        super().append(*args, **kargs)
+        super().append(*args, **kwargs)
         self._reassign_field()
 
-    def __setitem__(self, *args, **kargs):
+    def __setitem__(self, *args, **kwargs):
         """Set item.
 
         Args:
             args: The args passed.
-            kargs: The kwargs passed.
+            kwargs: The kwargs passed.
         """
-        super().__setitem__(*args, **kargs)
+        super().__setitem__(*args, **kwargs)
         self._reassign_field()
 
-    def __delitem__(self, *args, **kargs):
+    def __delitem__(self, *args, **kwargs):
         """Delete item.
 
         Args:
             args: The args passed.
-            kargs: The kwargs passed.
+            kwargs: The kwargs passed.
         """
-        super().__delitem__(*args, **kargs)
+        super().__delitem__(*args, **kwargs)
         self._reassign_field()
 
-    def clear(self, *args, **kargs):
+    def clear(self, *args, **kwargs):
         """Remove all item from the list.
 
         Args:
             args: The args passed.
-            kargs: The kwargs passed.
+            kwargs: The kwargs passed.
         """
-        super().clear(*args, **kargs)
+        super().clear(*args, **kwargs)
         self._reassign_field()
 
-    def extend(self, *args, **kargs):
+    def extend(self, *args, **kwargs):
         """Add all item of a list to the end of the list.
 
         Args:
             args: The args passed.
-            kargs: The kwargs passed.
+            kwargs: The kwargs passed.
         """
-        super().extend(*args, **kargs)
+        super().extend(*args, **kwargs)
         self._reassign_field() if hasattr(self, "_reassign_field") else None
 
-    def pop(self, *args, **kargs):
+    def pop(self, *args, **kwargs):
         """Remove an element.
 
         Args:
             args: The args passed.
-            kargs: The kwargs passed.
+            kwargs: The kwargs passed.
         """
-        super().pop(*args, **kargs)
+        super().pop(*args, **kwargs)
         self._reassign_field()
 
-    def remove(self, *args, **kargs):
+    def remove(self, *args, **kwargs):
         """Remove an element.
 
         Args:
             args: The args passed.
-            kargs: The kwargs passed.
+            kwargs: The kwargs passed.
         """
-        super().remove(*args, **kargs)
+        super().remove(*args, **kwargs)
+        self._reassign_field()
+
+
+class PCDict(dict):
+    """A custom dict that pynecone can detect its mutation."""
+
+    def __init__(
+        self,
+        original_dict: dict,
+        reassign_field: Callable = lambda _field_name: None,
+        field_name: str = "",
+    ):
+        """Initialize PCList.
+
+        Args:
+            original_dict (Dict): The original dict
```

Comment:
We can remove the types from the docstrings since we use type annotations

Reply:
ah right, didn't have time to finish this up, I'll do that as soon as I can :)